recipes/intel_install.rb

```diff
@@ -17,6 +17,20 @@
 
 return unless node['conditions']['intel_hpc_platform_supported'] && node['cfncluster']['enable_intel_hpc_platform'] == 'true'
 
+# Enforce keepcache=True for dnf, else downloaded packages will be removed after any successful install
+bash "configure dnf caching" do
+  cwd node['cfncluster']['sources_dir']
+  code <<-DNF_CACHE
```

Comment:
Let's use [`replace_or_add`](https://github.com/sous-chefs/line/blob/master/documentation/resources/replace_or_add.md) resource:
```ruby
replace_or_add "configure dnf caching" do
  path "/etc/dnf/dnf.conf"
  pattern "keepcache.*"
  line "keepcache=True"
end
```

This resource already has `ignore_missing` set to `true` by default so it's safe for non Centos8 systems too.